src/fileimpl.cpp

```diff
@@ -63,27 +63,62 @@ sectionSubReader(const FileReader& zimReader, const std::string& sectionName,
 #endif
 }
 
+struct NoDelete
+{
+  template<class T> void operator()(T*) {}
+};
+
+std::shared_ptr<Reader>
+makeFileReader(std::shared_ptr<const FileCompound> zimFile, offset_t offset, zsize_t size)
+{
+  if (zimFile->fail())
+    return nullptr;
+  else if ( zimFile->is_multiPart() ) {
+    ASSERT(offset.v, ==, 0u);
+    ASSERT(size, ==, zimFile->fsize());
+    return std::make_shared<MultiPartFileReader>(zimFile);
+  } else {
+    const auto& firstAndOnlyPart = zimFile->begin()->second;
+    const FileReader::FileHandle fh(&firstAndOnlyPart->fhandle(), NoDelete());
```

Comment:
Shouldn't we make `FilePart` manage a `std::shared_ptr<FS::FD>` instead of a plain `FS::FD` ?
This would allow us to drop this `NoDelete` shared ptr.

Reply:
That's definitely a safer solution.